lisa/tools/xfstests.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from pathlib import PurePath
+from typing import cast
+
+from lisa.executable import Tool
+from lisa.operating_system import Debian, Fedora, Posix
+
+from .git import Git
+from .make import Make
+
+
+class Xfstests(Tool):
+    repo = "https://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git"
+
+    @property
+    def command(self) -> str:
+        return "xfstests"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install_dep(self) -> None:
+        posix_os: Posix = cast(Posix, self.node.os)
+        tool_path = self.get_tool_path()
+        self.node.shell.mkdir(tool_path, exist_ok=True)
+        git = self.node.tools[Git]
+        git.clone(self.repo, tool_path)
+        # install dependency packages
+        if isinstance(self.node.os, Fedora):
+            posix_os.install_packages(
```

Comment:
This list is too long. Some dependent packages may be included automatically. We can let package system to include them.

Reply:
How do we know the top package name? then omit its dependent ones?

Reply:
we should figure out, which ones are direct used. Others should be included by package system like apt, yum.

Reply:
Anyway, put them into variables, and not put them in code. It makes the code is harder to read, because they are too long.